### PR TITLE
feat: inject OTEL_SERVICE_NAME and OTEL_RESOURCE_ATTRIBUTES via webhook

### PR DESCRIPTION
**Pull request by blumamir**
For dotnet and OSS java which currently relies on the resource to inject the device id in odiglet, and then resolves it from kublet to the desired service name

### CODE REVIEW

**Review by edeNFed — APPROVED** [review]
💯

**Comment by blumamir** [comment]
> Can we remove `collector/processors/odigosresourcenameprocessor` after this PR?

If we remove the processor, then after the first upgrade, there will still be pods that started without the webhook and if the processor is not there, the service name will show up as UUID. I suggest to keep it around for some-time until we fill comfortable that everyone has already upgraded, and then remove the processor

File: instrumentor/controllers/instrumentationdevice/pods_webhook.go [hunk 1]
```diff
@@ -5,15 +5,31 @@ import (
...
 	"strings"
 
-	common "github.com/odigos-io/odigos/common"
-	"github.com/odigos-io/odigos/k8sutils/pkg/consts"
+	"github.com/odigos-io/odigos/common"
+	k8sconsts "github.com/odigos-io/odigos/k8sutils/pkg/consts"
+	containerutils "github.com/odigos-io/odigos/k8sutils/pkg/container"
+	"github.com/odigos-io/odigos/k8sutils/pkg/workload"
+	"go.opentelemetry.io/otel/attribute"
+	semconv "go.opentelemetry.io/otel/semconv/v1.4.0"
```
[the remark this answers is not in the document]
Comment (blumamir) [reply instrumentor/controllers/instrumentationdevice/pods_webhook.go]: good catch.
updated t0 1.26.0 which is what we use in other places in the codebase

**Review by RonFed — APPROVED** [review]

**Comment by damemi** [comment]
@blumamir I think there is a bug in older versions of k8s <1.24 the pod namespace is not always set in an admission webhook (https://github.com/kubernetes/website/issues/30574, see https://github.com/kubernetes/website/issues/30574#issuecomment-974701582). I'm hitting this while testing https://github.com/odigos-io/odigos/pull/2105 on 1.20 and 1.23 where the instrumentor gives the following errors trying to reconcile a new service name:

```
2025-01-03T20:38:36Z	ERROR	admission	failed to get workload object from cache. cannot check for workload annotation. using workload name as OTEL_SERVICE_NAME	{"webhookGroup": "", "webhookKind": "Pod", "Pod": {"name":"","namespace":"default"}, "namespace": "default", "name": "", "resource": {"group":"","version":"v1","resource":"pods"}, "user": "system:serviceaccount:kube-system:replicaset-controller", "requestID": "8e9331f3-b5fa-4334-a238-a755a32d7b62", "error": "Deployment.apps \"inventory\" not found"}
github.com/odigos-io/odigos/instrumentor/controllers/instrumentationdevice.(*PodsWebhook).getServiceNameForEnv
	/workspace/instrumentor/controllers/instrumentationdevice/pods_webhook.go:67
github.com/odigos-io/odigos/instrumentor/controllers/instrumentationdevice.(*PodsWebhook).Default
	/workspace/instrumentor/controllers/instrumentationdevice/pods_webhook.go:46
sigs.k8s.io/controller-runtime/pkg/webhook/admission.(*defaulterForType).Handle
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.19.3/pkg/webhook/admission/defaulter_custom.go:80
sigs.k8s.io/controller-runtime/pkg/webhook/admission.(*Webhook).Handle
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.19.3/pkg/webhook/admission/webhook.go:181
sigs.k8s.io/controller-runtime/pkg/webhook/admission.(*Webhook).ServeHTTP
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.19.3/pkg/webhook/admission/http.go:119
...
```

Looking into alternatives to get the namespace to fix

**Pull request merged**